public/app/features/alerting/unified/rule-list/filter/useSavedSearches.ts

```diff
@@ -0,0 +1,380 @@
+ *
+ * Features:
+ * - Persists saved searches to UserStorage (syncs across devices)
+ * - Validates data schema on load (filters invalid entries)
+ * - Validates name uniqueness (case-insensitive)
+ * - Tracks analytics for all actions
+ * - Shows error notifications on storage failures
+ * - Provides auto-apply logic for default search on navigation
+ * - Per-user session tracking to handle logout/login scenarios
+ *
+ * @example
+ * ```tsx
+ * const { savedSearches, saveSearch, isLoading } = useSavedSearches();
+ *
+ * // Save current search
+ * const error = await saveSearch('My Search', currentQuery);
+ * if (error) {
+ *   // Handle validation error
+ * }
+ *
+ * // Auto-apply default search on mount
+ * useEffect(() => {
+ *   const defaultSearch = getAutoApplySearch();
+ *   if (defaultSearch) {
+ *     applySearch(defaultSearch);
+ *   }
+ * }, []);
+ * ```
+ */
+export function useSavedSearches(): UseSavedSearchesResult {
+  const [savedSearches, setSavedSearches] = useState<SavedSearch[]>([]);
+  const notifyApp = useAppNotification();
+
+  // Track whether we've already loaded to prevent double-loading
+  const hasLoadedRef = useRef(false);
+  // Track whether this is a fresh navigation (not refresh)
+  const isNavigationRef = useRef<boolean | null>(null);
+
+  // Async loader function for saved searches
+  const loadSavedSearches = useCallback(async (): Promise<SavedSearch[]> => {
+    const stored = await userStorage.getItem(SAVED_SEARCHES_STORAGE_KEY);
+    if (!stored) {
+      return [];
+    }
+
+    const parsed = JSON.parse(stored);
+    return validateSavedSearches(parsed);
+  }, []);
+
+  // Use useAsync for loading state management
+  const [{ execute: executeLoad }, loadState] = useAsync(loadSavedSearches, []);
+  const isLoading = isLoadingState(loadState) || isUninitialized(loadState);
+
+  // Load saved searches from storage on mount
+  useEffect(() => {
+    if (hasLoadedRef.current) {
+      return;
+    }
+    hasLoadedRef.current = true;
+
+    const sessionKey = getSessionVisitedKey();
+
+    // Determine if this is a navigation or refresh
+    // If session storage has the visited flag, this is a refresh
+    const wasVisited = sessionStorage.getItem(sessionKey);
+    isNavigationRef.current = !wasVisited;
+
+    // Set the visited flag for future checks
+    sessionStorage.setItem(sessionKey, 'true');
+
+    // Load from UserStorage using useAsync
+    executeLoad()
+      .then((validated) => {
+        setSavedSearches(validated);
+      })
+      .catch((error) => {
+        console.error('Failed to load saved searches from storage:', error);
+        notifyApp.error(
+          t('alerting.saved-searches.error-load-title', 'Failed to load saved searches'),
+          t(
+            'alerting.saved-searches.error-load-description',
+            'Your saved searches could not be loaded. Please try refreshing the page.'
+          )
+        );
+      });
```

Comment:
I think all of these should be handled in `loadSavedSearches` function. We shouldn't mix async/await and then/catch